src/components/modal/ModalComposite.tsx

```diff
@@ -99,6 +106,16 @@ export class ModalComposite extends React.PureComponent<IModalCompositeProps & P
         window.clearTimeout(this.timeoutId);
     }
 
+    private onRequestClose() {
+        if (this.props.validateShouldNavigate) {
+            if (this.props.validateShouldNavigate(this.props.isDirty)) {
+                this.props.onClose();
+            }
+        } else {
+            callIfDefined(this.props.onClose);

```

Comment:
You check if onClose is defined here, but not in the previous if, is this expected?